driver/terraform.go

```diff
@@ -558,7 +558,15 @@ func (tf *Terraform) initTaskTemplate() error {
 	return nil
 }
 
-func (tf *Terraform) setNotifier(tmpl templates.Template, serviceCount int) {
+func (tf *Terraform) setNotifier(tmpl templates.Template) {
+	serviceCount := len(tf.task.Services())
+	if cond, ok := tf.task.Condition().(*config.ServicesConditionConfig); ok {
+		serviceCount += len(cond.Names)
+	}
+	if si, ok := tf.task.SourceInput().(*config.ServicesSourceInputConfig); ok {
+		serviceCount += len(si.Names)
+	}
```

Comment:
I'm losing track of the possible combinations with the deprecation. Can you double check me if I'm understanding the intent correctly?

`task.services` cannot be configured alongside [services condition](https://github.com/hashicorp/consul-terraform-sync/blob/f2b798f835da2f43a5e7416b7c00b56177a34c6f/config/task.go#L523) nor [services source input](https://github.com/hashicorp/consul-terraform-sync/blob/f2b798f835da2f43a5e7416b7c00b56177a34c6f/config/task.go#L552). If this is true, then the `+=`s are can be confusing without this context and can be interpreted that overlap is possible, and potentially double counting service names.

Could you replace it with overriding `serviceCount` instead?

```suggestion
	serviceCount := len(tf.task.Services())
	if cond, ok := tf.task.Condition().(*config.ServicesConditionConfig); ok {
		serviceCount = len(cond.Names)
	}
	if si, ok := tf.task.SourceInput().(*config.ServicesSourceInputConfig); ok {
		serviceCount = len(si.Names)
	}
```

Reply:
I was thinking about how we've talked about potentially supporting users to configure both `source_input "services"` and `condition "services"` down the line. This was trying to leave open this possibility but its over-simplified and, to your point, it could potentially could lead to double-counting of services.

Thanks for catching, I updated to your suggestion and added a comment.

Reply:
Ahh that convo is ringing a bell now, thanks for the reminder. 👍 on changing it based on the double count.